src/aaz_dev/cli/controller/portal_cli_generator.py

```diff
@@ -0,0 +1,359 @@
+from command.model.configuration import CMDCommand, CMDCommandExample
+from command.model.specs import CMDSpecsCommand, CMDSpecsCommandVersion
+from utils import exceptions
+from utils.config import Config
+import os, json, re
+
+
+class PortalCliGenerator:
+    COMMAND_ROOT_NAME = "az"
+    DOC_ROOT_NAME = "https://docs.microsoft.com/cli/azure"
+
+    # az change-analysis list --star-ttime bbb --endtime aaa -resource-group aaa
+    PARA_REG_PATTERN = re.compile(r" --?([a-zA-Z0-9\-]+) ?")
+
+    # /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}...
+    # /providers/Microsoft.Network/dnsResolvers/{dnsResolverName}/inboundEndpoints/{inboundEndpointName}
+    RESOURCE_PATH_PATTERN = re.compile(r"/([a-zA-Z0-9\-]+)/?")
+
+    # /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}...
+    # /providers/Microsoft.OperationalInsights/workspaces/{workspaceName}...
+    # /providers/Microsoft.SecurityInsights/incidents/{incidentId}
+    PROVIDERS_LIST = ['Microsoft', 'Nginx']
+    def __init__(self):
+        pass
+
+    def fill_resource_type(self, cmd_portal_info, cmd_cfg, leaf):
+        """
+        path: ....Microsoft.Network/dnsForwardingRulesets/{dnsForwardingRulesetName}/forwardingRules/{forwardingRuleName}
+        resource_type: dnsForwardingRulesets/forwardingRules
+        path: ....Microsoft.Network/dnsForwardingRulesets/{dnsForwardingRulesetName}/forwardingRules -> list cmd
+        resource_type: dnsForwardingRulesets
+        """
+        if cmd_cfg.resources is None:
+            raise exceptions.ResourceNotFind("Resource not exist")
+        resource = cmd_cfg.resources[-1]
+        if resource.rp_name is None:
+            raise exceptions.ResourceNotFind("Resource rp_name not exist")
+        if resource.swagger_path is None:
+            raise exceptions.ResourceNotFind("Resource swagger_path not exist")
+        rp_name = resource.rp_name
+        swagger_path = resource.swagger_path
+        cmd_portal_info['rp_name'] = rp_name
+        swagger_provider_ind = swagger_path.rfind(rp_name)
+        provider_ind = 0
+        while swagger_provider_ind == -1 and provider_ind < len(self.PROVIDERS_LIST):
+            swagger_provider_ind = swagger_path.rfind(self.PROVIDERS_LIST[provider_ind])
+            provider_ind += 1
+        if swagger_provider_ind == -1:
+            print("please check cmd " + " ".join(leaf.names) + " resource id: " + swagger_path)
+            return
+        resource_paths = re.finditer(self.RESOURCE_PATH_PATTERN, swagger_path[swagger_provider_ind:])
+        folder_end_ind = swagger_path[swagger_provider_ind:].rfind("{", )
```

Comment:
switch these two lines and the condition check in line 58 can be removed
```suggestion
        folder_end_ind = swagger_path[swagger_provider_ind:].rfind("{", )
        resource_paths = re.finditer(self.RESOURCE_PATH_PATTERN, swagger_path[swagger_provider_ind:folder_end_ind])
```